src/main/scala/io/iohk/ethereum/blockchain/sync/FastSyncBlockBodiesHandler.scala

```diff
@@ -0,0 +1,65 @@
+package io.iohk.ethereum.blockchain.sync
+
+import akka.event.LoggingAdapter
+import akka.util.ByteString
+import io.iohk.ethereum.blockchain.sync.BlacklistSupport.BlackListId
+import io.iohk.ethereum.blockchain.sync.FastSyncBlockBodiesValidator.BlockBodyValidationResult
+import io.iohk.ethereum.network.Peer
+import io.iohk.ethereum.network.p2p.messages.PV62.BlockBody
+import io.iohk.ethereum.utils.Config.SyncConfig
+import org.bouncycastle.util.encoders.Hex
+
+import scala.concurrent.duration.FiniteDuration
+
+trait FastSyncBlockBodiesHandler extends FastSyncBlockBodiesValidator {
+
+  def syncConfig: SyncConfig
+  def log: LoggingAdapter
+
+  def handleBlockBodies(
+    peer: Peer,
+    requestedHashes: Seq[ByteString],
+    blockBodies: Seq[BlockBody],
+    handlerState: FastSyncHandlerState,
+    blacklist: (BlackListId, FiniteDuration, String) => Unit,
+    updateBestBlock: (Seq[ByteString]) => Unit
+  ): FastSyncHandlerState = {
+    if (blockBodies.isEmpty) {
```

Comment:
Methods take `FastSyncHandlerState` and return `FastSyncHandlerState`. Next step is State monad ;P